### ISSUE TITLE
Fix TF deployment issue / terraform-stats step

### ISSUE THREAD

**Issue by devopsmatt** [CDCgov/prime-reportstream#16819]
Context:
https://github.com/CDCgov/prime-reportstream/actions/runs/12359553628
https://github.com/CDCgov/prime-reportstream/actions/runs/12359553628/job/34492594460#step:4:45
https://github.com/CDCgov/prime-reportstream/actions/runs/12359553628/workflow

Ask:
Figure out why this fails, what is the value of 'terraform-stats' and consider removing that action from the TF deployment workflow if the value is low.

DoD: 
TF can be deployed via https://github.com/CDCgov/prime-reportstream/actions/runs/12359553628/workflow

**Comment by devopsmatt** [comment]
May resolve #16534

**Comment by emvaldes** [comment]
There are a few issues here that need to be addressed.
I will be working on this issue.

**Comment by emvaldes** [comment]
@devopsmatt: Please, could you address the ticket creation issue and establish the request?
It's known that the current Service Now capabilities are limited to only grant visibility to the issue-requester and you should be the one having full visibility on this so that others at the management level can inquire you about it and you can then keep them informed in full details.

**Comment by devopsmatt** [comment]
@emvaldes please open a ticket with Cloud Services and work with them to resolution.  If you need me in any working sessions, please pull me in.

**Comment by emvaldes** [comment]
A ticket was submitted by @devopsmatt to the CDC Cloud support team.
**Submitted** : 01/02/2025 17:59:28
**Request Number** : REQ0399128

I am awaiting for updates to this project.

**Comment by emvaldes** [comment]
`Run /home/runner/work/_actions/josiahsiegel/terraform-stats/68b8cbe42c494333fbf6f8d90ac86da1fb69dcc2/lib/tf_stats.sh \

  with data.azurerm_key_vault_secret.OKTA_authKey,
  on data.tf line 75, in data "azurerm_key_vault_secret" "OKTA_authKey":
  75: data "azurerm_key_vault_secret" "OKTA_authKey" {


Error: making Read request on Azure KeyVault Secret tf-caller-ip-addresses: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-keyvault;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}

  with data.azurerm_key_vault_secret.caller_ip_addresses,
  on data.tf line 82, in data "azurerm_key_vault_secret" "caller_ip_addresses":
  82: data "azurerm_key_vault_secret" "caller_ip_addresses" {


Error: making Read request on Azure KeyVault Secret functionapp-RS-OKTA-ClientId: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}

  with data.azurerm_key_vault_secret.RS_OKTA_clientId,
  on data.tf line 86, in data "azurerm_key_vault_secret" "RS_OKTA_clientId":
  86: data "azurerm_key_vault_secret" "RS_OKTA_clientId" {


Error: making Read request on Azure KeyVault Secret functionapp-RS-OKTA-authkey: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}

  with data.azurerm_key_vault_secret.RS_OKTA_authKey,
  on data.tf line 92, in data "azurerm_key_vault_secret" "RS_OKTA_authKey":
  92: data "azurerm_key_vault_secret" "RS_OKTA_authKey" {


Error: making Read request on Azure KeyVault Secret functionapp-cdctiautomated: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}

  with data.azurerm_key_vault_secret.cdctiautomated_sa,
  on data.tf line 98, in data "azurerm_key_vault_secret" "cdctiautomated_sa":
  98: data "azurerm_key_vault_secret" "cdctiautomated_sa" {


Error: making Read request on Azure KeyVault Secret hhsprotect-ip-ingress: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-keyvault;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}

  with module.storage.data.azurerm_key_vault_secret.hhsprotect_ip_ingress,
  on ../../modules/storage/data.tf line 3, in data "azurerm_key_vault_secret" "hhsprotect_ip_ingress":
   3: data "azurerm_key_vault_secret" "hhsprotect_ip_ingress" {


Error: making Read request on Azure KeyVault Secret cyberark-ip-ingress: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-keyvault;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}

  with module.storage.data.azurerm_key_vault_secret.cyberark_ip_ingress,
  on ../../modules/storage/data.tf line 8, in data "azurerm_key_vault_secret" "cyberark_ip_ingress":
   8: data "azurerm_key_vault_secret" "cyberark_ip_ingress" {

Error: Process completed with exit code 1.`

**Comment by emvaldes** [comment]
The issue here with the Terraform Service Principal yielding all these errors does not seem to be related at all with permissions to be granted to the PRIME-ReportStream Service Principal that was recently created because they are separate accounts for different purposes.
It has to do with the fact that the existing Terraform Service Principal certificate/secret has also expired (same problem we faced last time).

The question here is:
Should we go ahead and create a new Service Principal or should we try to fix the one that is currently setup but it has credentials that are no longer valid (expired: Certificate/Secret)?

**Comment by emvaldes** [comment]
This is my suggestion:

We should create a new Service Principal (as per the previous approach to get the issues fixed) but this time make sure we know exactly what those permissions are. If we keep the one that is now, we might be ok (short/quick fix) but it will not have a proper naming convention.

So, we could use something like:

* PRIME-ReportStream-Terraform
* PRIME-ReportStream--Terraform-Automation (I wouldn't recommend this one but it's in line with the existing naming convention).

**Comment by emvaldes** [comment]
The reason for suggesting/recommending to create a new Service Principal is because of all the weird things we were able to discover the certificate that was setup for the previous iteration of the now active PRIME-ReportStream Service Principal. So, to make sure we are in compliance with what the CDC Cloud support team does, I would suggest we ask for a new one to be created.

We need to follow the process the CDC Cloud support team explained they have set to be used. For that, we will need to make a new ticket that will take care of that.

**Comment by emvaldes** [comment]
The error we get at the Azure console is:
A certificate or secret has expired. Create a new one ->

**Comment by emvaldes** [comment]
There are also other certificates (no house-keeping at all):

* **rbac** -> 12/3/2021
* **rbac** -> 12/3/2022
* **Renewal** -> 12/5/2023
* **terraform-automation**?????? -> 11/26/2024 (I am assuming this is the one that broke)
* **terraform-automation**?????? -> 11/26/2025 (This one was created on November 26th 2024 but who did that?)

**Comment by emvaldes** [comment]
The request to create the Azure Service Principal with ID PRIME-ReportStream is now completed.

Note: We are pending on the resolution for the Service NOW ticket to configure this SP with the Cloud Application Administrator role (part of the default Azure roles). Ticket ID = RITM0407552

**Comment by emvaldes** [comment]
``` console
devops: prime-reportstream (main %=) $ find . -type f | xargs -I {} egrep -Hni 'terraform-automation' {} ;
./operations/app/terraform/modules/init/key_vault.tf:109:  // terraform-automation 5ab367bf-df15-45af-a027-47f95f2c75d8
```

**Comment by emvaldes** [comment]
This is the same issue we encountered before. The result of improperly constructing these values making it an obscure process.
``` hcl
 104   │ resource "azurerm_key_vault_access_policy" "init_tf" {
 105   │   for_each = toset(["appconfig", "keyvault"])
 106   │ 
 107   │   key_vault_id = azurerm_key_vault.init[each.value].id
 108   │   tenant_id    = data.azurerm_client_config.current.tenant_id
 109   │   // terraform-automation 5ab367bf-df15-45af-a027-47f95f2c75d8
 110   │   object_id = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
 111   │ 
```

**Comment by emvaldes** [comment]
The following configurations need to be changed:

``` console
devops: prime-reportstream (main %=) $ find . -type f -name '*.tf' | xargs -I {} egrep -Hni '4d81288c-27a3-4df8-b776-c9da8e688bc7' {} ;
./operations/app/terraform/vars/demo/locals.tf:29:    terraform_object_id       = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
./operations/app/terraform/vars/prod/locals.tf:28:    terraform_object_id       = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
./operations/app/terraform/vars/staging/locals.tf:28:    terraform_object_id       = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
./operations/app/terraform/vars/test/locals.tf:28:    terraform_object_id       = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
./operations/app/terraform/modules/init/key_vault.tf:110:  object_id = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
devops: prime-reportstream (main %=) $ 
devops: prime-reportstream (main %=) $ find . -type f -name '*.tf' | xargs -I {} egrep -Hni '5ab367bf-df15-45af-a027-47f95f2c75d8' {} ;
./operations/app/terraform/modules/init/key_vault.tf:109:  // terraform-automation 5ab367bf-df15-45af-a027-47f95f2c75d8
```

**Comment by emvaldes** [comment]
We have an opportunity here to start remediating a significant flaw in the existing Terraform and GitHub Actions implementation. These hardcoded credentials should not be allowed or supported.

We need to start changing this approach for at least to use GitHub Environment settings.
I will work on a proposal to remediate all these issues.

Why does this matter?
* It does matter from the security posture regarding these types of hardcoded credentials.
* It's even more important because it creates a process dependency management that whenever a service account/principal needs to be changed, then operations and development get tied without logical reasoning.

**Comment by emvaldes** [comment]
Identifying the use of hardcoded subscription ids like this is an issue:
**subscription_id = "7d1e3999-6577-4cd5-b296-f518e5c8e677"**

``` console
$ find . -type f -name '*.tf' | xargs -I {} egrep -Hni 'subscription_id' {} ;

./operations/app/terraform/vars/demo/azure.tf:37:
./operations/app/terraform/vars/prod/azure.tf:28:
./operations/app/terraform/vars/staging/azure.tf:28:
./operations/app/terraform/vars/test/azure.tf:30:
```

There are instances where this is consumed via this abstraction and we should embrace this to make the code more adaptable and capable.
**subscription_id = data.azurerm_subscription.current.subscription_id**

``` console
./operations/app/terraform/modules/azure_dashboard/main.tf:13:
```

**Comment by emvaldes** [comment]
I have done some additional research and I have manage to get identified all the data that is hardcoded and should not be:

``` console
$ find . -type f -name '*.tf' | xargs -I {} egrep -Hn "^(.*)([[:alnum:]]{8})-([[:alnum:]]{4})\-([[:alnum:]]{4})\-([[:alnum:]]{4})\-([[:alnum:]]{12})(.*)" {} ;
```

``` console
./operations/app/terraform/vars/demo/locals.tf:29:    terraform_object_id       = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
./operations/app/terraform/vars/demo/locals.tf:30:    aad_object_keyvault_admin = "3c17896c-ff94-4298-a719-aaac248aa2c8"
./operations/app/terraform/vars/demo/locals.tf:31:    aad_group_postgres_admin  = "f94409a9-12b1-4820-a1b6-e3e0a4fa282d"
./operations/app/terraform/vars/demo/azure.tf:37:  subscription_id            = "7d1e3999-6577-4cd5-b296-f518e5c8e677"
./operations/app/terraform/vars/prod/locals.tf:28:    terraform_object_id       = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
./operations/app/terraform/vars/prod/locals.tf:29:    aad_object_keyvault_admin = "5c6a951e-a4c2-4890-b62c-0ed8179501bb"
./operations/app/terraform/vars/prod/locals.tf:30:    aad_group_postgres_admin  = "c4031f1f-229c-4a8a-b3b9-23bae9dbf197"
./operations/app/terraform/vars/prod/azure.tf:28:  subscription_id            = "7d1e3999-6577-4cd5-b296-f518e5c8e677"
./operations/app/terraform/vars/staging/locals.tf:28:    terraform_object_id       = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
./operations/app/terraform/vars/staging/locals.tf:29:    aad_object_keyvault_admin = "b35a2a63-aeb2-438c-913b-bebeb821adfe"
./operations/app/terraform/vars/staging/locals.tf:30:    aad_group_postgres_admin  = "c4031f1f-229c-4a8a-b3b9-23bae9dbf197"
./operations/app/terraform/vars/staging/azure.tf:28:  subscription_id            = "7d1e3999-6577-4cd5-b296-f518e5c8e677"
./operations/app/terraform/vars/test/locals.tf:28:    terraform_object_id       = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
./operations/app/terraform/vars/test/locals.tf:29:    aad_object_keyvault_admin = "3c17896c-ff94-4298-a719-aaac248aa2c8"
./operations/app/terraform/vars/test/locals.tf:30:    aad_group_postgres_admin  = "f94409a9-12b1-4820-a1b6-e3e0a4fa282d"
./operations/app/terraform/vars/test/azure.tf:30:  subscription_id            = "7d1e3999-6577-4cd5-b296-f518e5c8e677"
./operations/app/terraform/modules/init/key_vault.tf:77:  object_id = "52fe06f4-9717-4beb-9b7a-e05b6bdd2f0f"
./operations/app/terraform/modules/init/key_vault.tf:109:  // terraform-automation 5ab367bf-df15-45af-a027-47f95f2c75d8
./operations/app/terraform/modules/init/key_vault.tf:110:  object_id = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
./operations/app/terraform/modules/application_insights/alert_availability.tf:85:    "hidden-link:/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/${var.resource_group}/providers/Microsoft.Insights/components/${var.resource_prefix}-appinsights" = "Resource"
./operations/app/terraform/modules/application_insights/alert_availability.tf:110:    "hidden-link:/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/${var.resource_group}/providers/Microsoft.Insights/components/${var.resource_prefix}-appinsights" = "Resource"
./operations/app/terraform/modules/application_insights/alert_availability.tf:164:    "hidden-link:/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/${var.resource_group}/providers/Microsoft.Insights/components/${var.resource_prefix}-appinsights" = "Resource"
./operations/app/terraform/modules/front_door/main.tf:303:  object_id = "270e4d1a-12bd-4564-8a4b-c9de1bbdbe95"
```

**Comment by emvaldes** [comment]
The Azure Service Principal "PRIME-ReportStream-Terraform" is now fully setup and I will proceed to get it updated in the Terraform configurations.

**Comment by emvaldes** [comment]
A contributor light role was added to the following resource groups for the new SP:

`prime-data-hub-demo, 
prime-data-hub-demo1, 
prime-data-hub-demo2, 
prime-data-hub-demo3, 
prime-data-hub-prod, 
prime-data-hub-staging, 
prime-data-hub-test`

Setup access policies for the new Service Principal on the following key vaults was setup with the same permissions as with the original terraform-automation (original Azure Service Principal):

`pdhdemo1-appconfigs5m, 
pdhdemo1-keyvaults5m, 
pdhdemo2-appconfig1wu, 
pdhdemo2-keyvault1wu, 
pdhdemo3-appconfigjeo, 
pdhdemo3-keyvaultjeo, 
pdhprod-appconfig, 
pdhprod-keyvault, 
pdhstaging-appconfig, 
pdhstaging-keyvault, 
pdhtest-app-config, 
pdhtest-keyvault`

**Comment by emvaldes** [comment]
**Warnings**: After some investigation, the original Azure Service Principal (**terraform-automation**) had a contributor role, but it's at the **Subscription** level.
The team cannot grant either the **role** nor the **location** for that without approval from higher up.
They expressed their surprise that the original team was able to have those privs within the DMZ.

**Comment by emvaldes** [comment]
The following change is required (a formality as this is just a comment:

``` console
devops: prime-reportstream (main %=) $ find . -type f -name '*.tf' | xargs -I {} egrep -Hni '5ab367bf-df15-45af-a027-47f95f2c75d8' {} ;
./operations/app/terraform/modules/init/key_vault.tf:109:  // terraform-automation 5ab367bf-df15-45af-a027-47f95f2c75d8
```

**Comment by emvaldes** [comment]
The following change is required (a hardcoded value):

``` console
./operations/app/terraform/vars/demo/locals.tf:29:
terraform_object_id = "4d81288c-27a3-4df8-b776-c9da8e688bc7"

./operations/app/terraform/vars/prod/locals.tf:28:
terraform_object_id = "4d81288c-27a3-4df8-b776-c9da8e688bc7"

./operations/app/terraform/vars/staging/locals.tf:28:
terraform_object_id = "4d81288c-27a3-4df8-b776-c9da8e688bc7"

./operations/app/terraform/vars/test/locals.tf:28:
terraform_object_id = "4d81288c-27a3-4df8-b776-c9da8e688bc7"

./operations/app/terraform/modules/init/key_vault.tf:110:
object_id = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
```

**Comment by emvaldes** [comment]
I am pointing out that this information is also present in files as content outside the Terraform configurations.

``` console
./prime-router/docs/docs-deprecated/environment-provisioning.md:87:
"/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-data-hub-$env/providers/Microsoft.KeyVault/vaults/pdh$env-appconfigmt8/objectId/4d81288c-27a3-4df8-b776-c9da8e688bc7"

./prime-router/docs/docs-deprecated/environment-provisioning.md:91:
"/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-data-hub-$env/providers/Microsoft.KeyVault/vaults/pdh$env-keyvaultmt8/objectId/4d81288c-27a3-4df8-b776-c9da8e688bc7"
```

**Comment by emvaldes** [comment]
``` console
$ legacy_account='4d81288c-27a3-4df8-b776-c9da8e688bc7';
$ active_account='a58ee002-62c7-4a91-a2dc-4a837663aa00';
$ oIFS="${IFS}"; IFS=$'\n';
$ declare -a targets=($( find . -type f | xargs -I {} egrep -Hni "${legacy_account}" {} | awk -F':' '{print $1}' | sort -u ));
for target in ${targets[@]}; do
    echo -e "\nFile: ${change}";
    sed -i '' -e "s|${legacy_account}|{active_account}|g" ${target} ;
done;
```

**Comment by emvaldes** [comment]
``` console
devops: prime-reportstream (devsecops/emvaldes/terraform-service-principal *%) $ git diff ;
diff --git a/operations/app/terraform/modules/init/key_vault.tf b/operations/app/terraform/modules/init/key_vault.tf
index d8c1756ed..36b69673f 100644
--- a/operations/app/terraform/modules/init/key_vault.tf
+++ b/operations/app/terraform/modules/init/key_vault.tf
@@ -107,7 +107,7 @@ resource "azurerm_key_vault_access_policy" "init_tf" {
   key_vault_id = azurerm_key_vault.init[each.value].id
   tenant_id    = data.azurerm_client_config.current.tenant_id
   // terraform-automation 5ab367bf-df15-45af-a027-47f95f2c75d8
-  object_id = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
+  object_id = "a58ee002-62c7-4a91-a2dc-4a837663aa00"
 
   key_permissions = [
     "Create",
diff --git a/operations/app/terraform/vars/demo/locals.tf b/operations/app/terraform/vars/demo/locals.tf
index 52e3441ef..72250bbb3 100644
--- a/operations/app/terraform/vars/demo/locals.tf
+++ b/operations/app/terraform/vars/demo/locals.tf
@@ -26,7 +26,7 @@ locals {
     tf_secrets_vault      = "pdh${local.init.environment}-keyvault${local.init.random_id}"
   }
   ad = {
-    terraform_object_id       = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
+    terraform_object_id       = "a58ee002-62c7-4a91-a2dc-4a837663aa00"
     aad_object_keyvault_admin = "3c17896c-ff94-4298-a719-aaac248aa2c8"
     aad_group_postgres_admin  = "f94409a9-12b1-4820-a1b6-e3e0a4fa282d"
   }
diff --git a/operations/app/terraform/vars/prod/locals.tf b/operations/app/terraform/vars/prod/locals.tf
index a64ee7953..78df2edc7 100644
--- a/operations/app/terraform/vars/prod/locals.tf
+++ b/operations/app/terraform/vars/prod/locals.tf
@@ -25,7 +25,7 @@ locals {
     tf_secrets_vault      = "pdh${local.init.environment}-keyvault"
   }
   ad = {
-    terraform_object_id       = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
+    terraform_object_id       = "a58ee002-62c7-4a91-a2dc-4a837663aa00"
     aad_object_keyvault_admin = "5c6a951e-a4c2-4890-b62c-0ed8179501bb"
     aad_group_postgres_admin  = "c4031f1f-229c-4a8a-b3b9-23bae9dbf197"
   }
diff --git a/operations/app/terraform/vars/staging/locals.tf b/operations/app/terraform/vars/staging/locals.tf
index 47066309e..3a221af96 100644
--- a/operations/app/terraform/vars/staging/locals.tf
+++ b/operations/app/terraform/vars/staging/locals.tf
@@ -25,7 +25,7 @@ locals {
     tf_secrets_vault      = "pdh${local.init.environment}-keyvault"
   }
   ad = {
-    terraform_object_id       = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
+    terraform_object_id       = "a58ee002-62c7-4a91-a2dc-4a837663aa00"
     aad_object_keyvault_admin = "b35a2a63-aeb2-438c-913b-bebeb821adfe"
     aad_group_postgres_admin  = "c4031f1f-229c-4a8a-b3b9-23bae9dbf197"
   }
diff --git a/operations/app/terraform/vars/test/locals.tf b/operations/app/terraform/vars/test/locals.tf
index b229fad36..8ff6b7112 100644
--- a/operations/app/terraform/vars/test/locals.tf
+++ b/operations/app/terraform/vars/test/locals.tf
@@ -25,7 +25,7 @@ locals {
     tf_secrets_vault      = "pdh${local.init.environment}-keyvault"
   }
   ad = {
-    terraform_object_id       = "4d81288c-27a3-4df8-b776-c9da8e688bc7"
+    terraform_object_id       = "a58ee002-62c7-4a91-a2dc-4a837663aa00"
     aad_object_keyvault_admin = "3c17896c-ff94-4298-a719-aaac248aa2c8"
     aad_group_postgres_admin  = "f94409a9-12b1-4820-a1b6-e3e0a4fa282d"
   }
diff --git a/prime-router/docs/docs-deprecated/environment-provisioning.md b/prime-router/docs/docs-deprecated/environment-provisioning.md
index 83d62558c..5b9d3c602 100644
--- a/prime-router/docs/docs-deprecated/environment-provisioning.md
+++ b/prime-router/docs/docs-deprecated/environment-provisioning.md
@@ -84,11 +84,11 @@ echo "init complete"
 # Import access polices that are shared with init and key_vault modules
 terraform -chdir=$path import -var-file=$env/env.tfvars.json \
 module.key_vault.azurerm_key_vault_access_policy.terraform_app_config_access_policy[0] \
-"/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-data-hub-$env/providers/Microsoft.KeyVault/vaults/pdh$env-appconfigmt8/objectId/4d81288c-27a3-4df8-b776-c9da8e688bc7"
+"/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-data-hub-$env/providers/Microsoft.KeyVault/vaults/pdh$env-appconfigmt8/objectId/a58ee002-62c7-4a91-a2dc-4a837663aa00"
 
 terraform -chdir=$path import -var-file=$env/env.tfvars.json \
  module.key_vault.azurerm_key_vault_access_policy.terraform_access_policy[0] \
-"/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-data-hub-$env/providers/Microsoft.KeyVault/vaults/pdh$env-keyvaultmt8/objectId/4d81288c-27a3-4df8-b776-c9da8e688bc7"
+"/subscriptions/7d1e3999-6577-4cd5-b296-f518e5c8e677/resourceGroups/prime-data-hub-$env/providers/Microsoft.KeyVault/vaults/pdh$env-keyvaultmt8/objectId/a58ee002-62c7-4a91-a2dc-4a837663aa00"
 
 for i in {1..3}; do \
 terraform -chdir=$path apply \
```

**Comment by emvaldes** [comment]
``` console
devops: prime-reportstream (devsecops/emvaldes/terraform-service-principal *%) $ git status ;
On branch devsecops/emvaldes/terraform-service-principal
Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   operations/app/terraform/modules/init/key_vault.tf
	modified:   operations/app/terraform/vars/demo/locals.tf
	modified:   operations/app/terraform/vars/prod/locals.tf
	modified:   operations/app/terraform/vars/staging/locals.tf
	modified:   operations/app/terraform/vars/test/locals.tf
	modified:   prime-router/docs/docs-deprecated/environment-provisioning.md

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	.github/actions/workflow-housekeeper/

no changes added to commit (use "git add" and/or "git commit -a")
```

**Comment by emvaldes** [comment]
``` console
devops: prime-reportstream (devsecops/emvaldes/terraform-service-principal *) $ git add . && git commit -m "Updating Terraform Azure Service Principal PRIME-ReportStream-Terraform";
[devsecops/emvaldes/terraform-service-principal 35d40af4a] Updating Terraform Azure Service Principal PRIME-ReportStream-Terraform
 6 files changed, 7 insertions(+), 7 deletions(-)
```

**Comment by emvaldes** [comment]
``` console
devops: prime-reportstream (devsecops/emvaldes/terraform-service-principal) $ git push --set-upstream origin devsecops/emvaldes/terraform-service-principal ;
Enumerating objects: 41, done.
Counting objects: 100% (41/41), done.
Delta compression using up to 16 threads
Compressing objects: 100% (21/21), done.
Writing objects: 100% (21/21), 1.56 KiB | 1.56 MiB/s, done.
Total 21 (delta 18), reused 0 (delta 0), pack-reused 0 (from 0)
remote: Resolving deltas: 100% (18/18), completed with 18 local objects.
remote: 
remote: Create a pull request for 'devsecops/emvaldes/terraform-service-principal' on GitHub by visiting:
remote:      https://github.com/CDCgov/prime-reportstream/pull/new/devsecops/emvaldes/terraform-service-principal
remote: 
To https://github.com/CDCgov/prime-reportstream.git
 * [new branch]          devsecops/emvaldes/terraform-service-principal -> devsecops/emvaldes/terraform-service-principal
branch 'devsecops/emvaldes/terraform-service-principal' set up to track 'origin/devsecops/emvaldes/terraform-service-principal'.
```

**Comment by emvaldes** [comment]
An issue was identified and now corrected where the update to the Terraform Service Principal was not correct.

**Comment by emvaldes** [comment]
Although this PR does not address these issues, it's important to mention that there are lots of Azure account related information that is hardcoded into the code-base. Something that should eventually be remediated.

We should not be supporting or enforcing such practices but instead do all we can to bring the code-base to an example of good standards to follow. Hardcoding anything in a Software Configuration Management is just not right regardless.

Instead of maintaining a bad practice, this is what we should be doing either via secrets or environment variables:
e.g.: In your Terraform modules, define variables without assigning default values.

``` hcl
variable "subscription_id" {
  description = "Azure Subscription ID"
  type        = string
}
```

e.g.: Enforcing the scope of these values as environment values that will be available to the Terraform engine during run-time. Making it a flexible and agnostic process.

``` hcl
jobs:
  deploy:
    runs-on: ubuntu-latest
    env:
      ARM_SUBSCRIPTION_ID: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
      ARM_CLIENT_ID: ${{ secrets.AZURE_CLIENT_ID }}
      ARM_CLIENT_SECRET: ${{ secrets.AZURE_CLIENT_SECRET }}
      ARM_TENANT_ID: ${{ secrets.AZURE_TENANT_ID }}
    steps:
      - name: Checkout code
        uses: actions/checkout@v4

      - name: Set up Terraform
        uses: hashicorp/setup-terraform@v3

      - name: Initialize Terraform
        run: terraform init

      - name: Apply Terraform configuration
        run: terraform apply -auto-approve
```

By not specifying the these values directly in the provider block, Terraform will reference the ARM_SUBSCRIPTION_ID environment variable during execution.
This method keeps your configuration clean and free from hardcoded sensitive data.

**Important Considerations**:

* Mandatory Subscription ID: As of AzureRM provider version 4.0.0, specifying the subscription_id is mandatory. If it's not provided in the provider block, Terraform will look for the ARM_SUBSCRIPTION_ID environment variable. Ensure this environment variable is set to avoid errors during plan or apply operations. 

* Environment Variable Precedence: The AzureRM provider gives precedence to environment variables for authentication details. If both the provider block and environment variables specify a subscription_id, the provider block value will take precedence. Therefore, ensure consistency between your environment variables and provider configurations.

* Security Best Practices: Storing sensitive information in environment variables, especially within CI/CD pipelines, is a common practice. However, always use secure storage solutions like GitHub Secrets to manage these variables, preventing unauthorized access.

By following this approach, you maintain a secure and flexible Terraform configuration, free from hardcoded sensitive information, and adaptable across different environments.

---

In Azure, identifiers such as subscription_id, client_id, and tenant_id are generally not considered highly sensitive on their own. However, when combined with other information, they can potentially be misused.

Potential Risks:

Information Disclosure: While these IDs alone don't grant access, they can reveal details about your Azure environment. For instance, a subscription_id can indicate the existence of a subscription, and a tenant_id can disclose your Azure Active Directory (AAD) tenant. 
[MICROSOFT LEARN](https://learn.microsoft.com/en-us/answers/questions/340725/is-subscription-id-personal-or-should-be-kept-priv?utm_source=chatgpt.com)

Targeted Attacks: Malicious actors could use these IDs to craft targeted phishing attacks or social engineering schemes, aiming to extract more sensitive information or credentials from your organization.

Credential Combination: If an attacker obtains these IDs along with other credentials, such as a client_secret or certificate, they could potentially authenticate as a service principal in your Azure environment, leading to unauthorized access. 
[SECUREWORKS](https://www.secureworks.com/research/abusing-azure-application-credentials-to-attack-supply-chains?utm_source=chatgpt.com)

Recommendations:

Limit Exposure: Avoid sharing these identifiers publicly or with untrusted parties. While they are not highly sensitive, minimizing their exposure reduces the risk of them being used in conjunction with other compromised information.

Secure Associated Credentials: Ensure that any credentials associated with these IDs, such as client_secrets or certificates, are stored securely using services like Azure Key Vault. This practice helps prevent unauthorized access even if the IDs are known. 
[MICROSOFT LEARN](https://learn.microsoft.com/en-us/answers/questions/1184166/what-are-values-that-are-considered-confidential-a?utm_source=chatgpt.com)

Monitor and Audit: Regularly monitor access logs and audit trails for any suspicious activities related to these IDs. Implementing Azure's monitoring tools can help detect and respond to potential threats promptly.

In summary, while subscription_id, client_id, and tenant_id are not inherently sensitive, they should still be handled with care to prevent potential misuse, especially when combined with other confidential information.

**Comment by emvaldes** [comment]
```
2025-01-14T19:33:51.9868404Z Error: making Read request on Azure KeyVault Secret sendgrid-password: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:51.9871976Z 
2025-01-14T19:33:51.9872214Z   with data.azurerm_key_vault_secret.sendgrid_password,
2025-01-14T19:33:51.9872914Z   on data.tf line 9, in data "azurerm_key_vault_secret" "sendgrid_password":
2025-01-14T19:33:51.9873618Z    9: data "azurerm_key_vault_secret" "sendgrid_password" {
2025-01-14T19:33:51.9874004Z 
2025-01-14T19:33:51.9874026Z 
2025-01-14T19:33:51.9878047Z Error: making Read request on Azure KeyVault Secret functionapp-postgres-user: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:51.9881758Z 
2025-01-14T19:33:51.9881987Z   with data.azurerm_key_vault_secret.postgres_user,
2025-01-14T19:33:51.9882631Z   on data.tf line 14, in data "azurerm_key_vault_secret" "postgres_user":
2025-01-14T19:33:51.9883280Z   14: data "azurerm_key_vault_secret" "postgres_user" {
2025-01-14T19:33:51.9883629Z 
2025-01-14T19:33:51.9910401Z 
2025-01-14T19:33:51.9914064Z Error: making Read request on Azure KeyVault Secret functionapp-postgres-pass: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:51.9917240Z 
2025-01-14T19:33:51.9917461Z   with data.azurerm_key_vault_secret.postgres_pass,
2025-01-14T19:33:51.9918119Z   on data.tf line 19, in data "azurerm_key_vault_secret" "postgres_pass":
2025-01-14T19:33:51.9918750Z   19: data "azurerm_key_vault_secret" "postgres_pass" {
2025-01-14T19:33:51.9919100Z 
2025-01-14T19:33:51.9919106Z 
2025-01-14T19:33:51.9922815Z Error: making Read request on Azure KeyVault Secret functionapp-postgres-readonly-user: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:51.9925983Z 
2025-01-14T19:33:51.9926238Z   with data.azurerm_key_vault_secret.postgres_readonly_user,
2025-01-14T19:33:51.9926947Z   on data.tf line 24, in data "azurerm_key_vault_secret" "postgres_readonly_user":
2025-01-14T19:33:51.9927657Z   24: data "azurerm_key_vault_secret" "postgres_readonly_user" {
2025-01-14T19:33:51.9928045Z 
2025-01-14T19:33:51.9936464Z 
2025-01-14T19:33:51.9943726Z Error: making Read request on Azure KeyVault Secret functionapp-postgres-readonly-pass: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:51.9950895Z 
2025-01-14T19:33:51.9951538Z   with data.azurerm_key_vault_secret.postgres_readonly_pass,
2025-01-14T19:33:51.9952394Z   on data.tf line 29, in data "azurerm_key_vault_secret" "postgres_readonly_pass":
2025-01-14T19:33:51.9953209Z   29: data "azurerm_key_vault_secret" "postgres_readonly_pass" {
2025-01-14T19:33:51.9953699Z 
2025-01-14T19:33:51.9953828Z 
2025-01-14T19:33:51.9957664Z Error: making Read request on Azure KeyVault Secret pagerduty-integration-url: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-keyvault;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:51.9961158Z 
2025-01-14T19:33:51.9961397Z   with data.azurerm_key_vault_secret.pagerduty_url,
2025-01-14T19:33:51.9962045Z   on data.tf line 39, in data "azurerm_key_vault_secret" "pagerduty_url":
2025-01-14T19:33:51.9962684Z   39: data "azurerm_key_vault_secret" "pagerduty_url" {
2025-01-14T19:33:51.9963043Z 
2025-01-14T19:33:51.9965567Z 
2025-01-14T19:33:51.9968932Z Error: keyvault.BaseClient#GetKey: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have keys get permission on key vault 'pdhstaging-keyvault;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:51.9972221Z 
2025-01-14T19:33:51.9972543Z   with data.azurerm_key_vault_key.pdh-2048-key,
2025-01-14T19:33:51.9973283Z   on data.tf line 44, in data "azurerm_key_vault_key" "pdh-2048-key":
2025-01-14T19:33:51.9973995Z   44: data "azurerm_key_vault_key" "pdh-2048-key" {
2025-01-14T19:33:51.9974477Z 
2025-01-14T19:33:51.9977182Z 
2025-01-14T19:33:51.9981336Z Error: making Read request on Azure KeyVault Secret slack-email: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-keyvault;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:51.9984414Z 
2025-01-14T19:33:51.9984670Z   with data.azurerm_key_vault_secret.slack_email_address,
2025-01-14T19:33:51.9985361Z   on data.tf line 49, in data "azurerm_key_vault_secret" "slack_email_address":
2025-01-14T19:33:51.9986049Z   49: data "azurerm_key_vault_secret" "slack_email_address" {
2025-01-14T19:33:51.9986430Z 
2025-01-14T19:33:51.9987677Z 
2025-01-14T19:33:51.9990924Z Error: making Read request on Azure KeyVault Secret chatops-slack-bot-token: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-keyvault;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:51.9992789Z 
2025-01-14T19:33:51.9992969Z   with data.azurerm_key_vault_secret.chatops_slack_bot_token,
2025-01-14T19:33:51.9993413Z   on data.tf line 54, in data "azurerm_key_vault_secret" "chatops_slack_bot_token":
2025-01-14T19:33:51.9993842Z   54: data "azurerm_key_vault_secret" "chatops_slack_bot_token" {
2025-01-14T19:33:51.9994079Z 
2025-01-14T19:33:51.9994381Z 
2025-01-14T19:33:51.9997063Z Error: making Read request on Azure KeyVault Secret chatops-slack-app-token: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-keyvault;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:51.9998989Z 
2025-01-14T19:33:51.9999164Z   with data.azurerm_key_vault_secret.chatops_slack_app_token,
2025-01-14T19:33:51.9999600Z   on data.tf line 59, in data "azurerm_key_vault_secret" "chatops_slack_app_token":
2025-01-14T19:33:52.0000183Z   59: data "azurerm_key_vault_secret" "chatops_slack_app_token" {
2025-01-14T19:33:52.0000580Z 
2025-01-14T19:33:52.0000640Z 
2025-01-14T19:33:52.0003102Z Error: making Read request on Azure KeyVault Secret chatops-github-token: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-keyvault;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:52.0004908Z 
2025-01-14T19:33:52.0005053Z   with data.azurerm_key_vault_secret.chatops_github_token,
2025-01-14T19:33:52.0005460Z   on data.tf line 64, in data "azurerm_key_vault_secret" "chatops_github_token":
2025-01-14T19:33:52.0005865Z   64: data "azurerm_key_vault_secret" "chatops_github_token" {
2025-01-14T19:33:52.0006086Z 
2025-01-14T19:33:52.0006482Z 
2025-01-14T19:33:52.0008933Z Error: making Read request on Azure KeyVault Secret functionapp-OKTA-ClientId: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:52.0010868Z 
2025-01-14T19:33:52.0011091Z   with data.azurerm_key_vault_secret.OKTA_clientId,
2025-01-14T19:33:52.0011733Z   on data.tf line 69, in data "azurerm_key_vault_secret" "OKTA_clientId":
2025-01-14T19:33:52.0012374Z   69: data "azurerm_key_vault_secret" "OKTA_clientId" {
2025-01-14T19:33:52.0012734Z 
2025-01-14T19:33:52.0013019Z 
2025-01-14T19:33:52.0016574Z Error: making Read request on Azure KeyVault Secret functionapp-OKTA-authkey: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:52.0019718Z 
2025-01-14T19:33:52.0019960Z   with data.azurerm_key_vault_secret.OKTA_authKey,
2025-01-14T19:33:52.0020737Z   on data.tf line 75, in data "azurerm_key_vault_secret" "OKTA_authKey":
2025-01-14T19:33:52.0021373Z   75: data "azurerm_key_vault_secret" "OKTA_authKey" {
2025-01-14T19:33:52.0021656Z 
2025-01-14T19:33:52.0021667Z 
2025-01-14T19:33:52.0023676Z Error: making Read request on Azure KeyVault Secret tf-caller-ip-addresses: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-keyvault;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:52.0025463Z 
2025-01-14T19:33:52.0025775Z   with data.azurerm_key_vault_secret.caller_ip_addresses,
2025-01-14T19:33:52.0026296Z   on data.tf line 82, in data "azurerm_key_vault_secret" "caller_ip_addresses":
2025-01-14T19:33:52.0026702Z   82: data "azurerm_key_vault_secret" "caller_ip_addresses" {
2025-01-14T19:33:52.0026927Z 
2025-01-14T19:33:52.0027616Z 
2025-01-14T19:33:52.0029737Z Error: making Read request on Azure KeyVault Secret functionapp-RS-OKTA-ClientId: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:52.0032442Z 
2025-01-14T19:33:52.0032694Z   with data.azurerm_key_vault_secret.RS_OKTA_clientId,
2025-01-14T19:33:52.0033376Z   on data.tf line 86, in data "azurerm_key_vault_secret" "RS_OKTA_clientId":
2025-01-14T19:33:52.0034038Z   86: data "azurerm_key_vault_secret" "RS_OKTA_clientId" {
2025-01-14T19:33:52.0034406Z 
2025-01-14T19:33:52.0038771Z 
2025-01-14T19:33:52.0041617Z Error: making Read request on Azure KeyVault Secret functionapp-RS-OKTA-authkey: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:52.0043464Z 
2025-01-14T19:33:52.0043605Z   with data.azurerm_key_vault_secret.RS_OKTA_authKey,
2025-01-14T19:33:52.0044001Z   on data.tf line 92, in data "azurerm_key_vault_secret" "RS_OKTA_authKey":
2025-01-14T19:33:52.0044389Z   92: data "azurerm_key_vault_secret" "RS_OKTA_authKey" {
2025-01-14T19:33:52.0044601Z 
2025-01-14T19:33:52.0047220Z 
2025-01-14T19:33:52.0049827Z Error: making Read request on Azure KeyVault Secret functionapp-cdctiautomated: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-appconfig;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:52.0052189Z 
2025-01-14T19:33:52.0052347Z   with data.azurerm_key_vault_secret.cdctiautomated_sa,
2025-01-14T19:33:52.0053008Z   on data.tf line 98, in data "azurerm_key_vault_secret" "cdctiautomated_sa":
2025-01-14T19:33:52.0053480Z   98: data "azurerm_key_vault_secret" "cdctiautomated_sa" {
2025-01-14T19:33:52.0053753Z 
2025-01-14T19:33:52.0053756Z 
2025-01-14T19:33:52.0055751Z Error: making Read request on Azure KeyVault Secret hhsprotect-ip-ingress: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-keyvault;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:52.0057546Z 
2025-01-14T19:33:52.0057746Z   with module.storage.data.azurerm_key_vault_secret.hhsprotect_ip_ingress,
2025-01-14T19:33:52.0058504Z   on ../../modules/storage/data.tf line 3, in data "azurerm_key_vault_secret" "hhsprotect_ip_ingress":
2025-01-14T19:33:52.0059092Z    3: data "azurerm_key_vault_secret" "hhsprotect_ip_ingress" {
2025-01-14T19:33:52.0059321Z 
2025-01-14T19:33:52.0059324Z 
2025-01-14T19:33:52.0061467Z Error: making Read request on Azure KeyVault Secret cyberark-ip-ingress: keyvault.BaseClient#GetSecret: Failure responding to request: StatusCode=403 -- Original Error: autorest/azure: Service returned an error. Status=403 Code="Forbidden" Message="The user, group or application 'appid=***;oid=ac21f8dc-f2ee-4a56-bd88-51a5ae230dd8;iss=https://sts.windows.net/***/' does not have secrets get permission on key vault 'pdhstaging-keyvault;location=eastus'. For help resolving this issue, please see https://go.microsoft.com/fwlink/?linkid=2125287" InnerError={"code":"AccessDenied"}
2025-01-14T19:33:52.0063256Z 
2025-01-14T19:33:52.0063447Z   with module.storage.data.azurerm_key_vault_secret.cyberark_ip_ingress,
2025-01-14T19:33:52.0063977Z   on ../../modules/storage/data.tf line 8, in data "azurerm_key_vault_secret" "cyberark_ip_ingress":
2025-01-14T19:33:52.0064442Z    8: data "azurerm_key_vault_secret" "cyberark_ip_ingress" {
```
